source/vibe/web/rest.d

```diff
@@ -370,7 +370,14 @@ class RestInterfaceClient(I) : I
 
 			URL url = m_baseURL;
 
-			if (name.length) url ~= Path(name);
+			if (name.length)
+			{
+				string sep = "";
+				if (name[0] != '/' && (!url.pathString.length || url.pathString[0] != '/'))
+					sep = "/";
+				url.pathString = url.pathString ~ sep ~ name;
```

Comment:
Path - is generally a complex topic.
I think there was a fundamental error to say that non-encoded path part of an URL and OS file path is the same thing. We may say that the Path in URL contains only encoded URL parts.
For example `localhost://foo bar/some thing.html`
Now considered that `.path` of this URL contains two parts: `foo bar` and `some thing.html`
However URL path parts may contain _any_ characters if it is properly encoded.
Moreover, `Path` can't know anything about url encoding, because it represents a common path, not only url path.
It is good idea to say that `assert(URL("localhost://foo bar/some thing.html").path == Path("foo%20bar/some%20thing.html"));`, however it will be break change, now, because in this case, `url.path = Path("foo bar/some thing.html")` should rise an error (Path should be already encoded).
We may add special type `struct UrlPath`, which inherited from `Path` via `alias this` and knows about encoding. After that, we may deprecate all `URL` methods, which takes or returns `Path`.